packages/bathymetry/src/bathy.maker.ts

```diff
@@ -82,17 +85,24 @@ export class BathyMaker {
         logger.info({ version: gdalVersion }, 'GdalVersion');
 
         const promises = [];
+        let extent: Bounds | null = null;
         for (let x = 0; x < tmsZoom.matrixWidth; x++) {
             for (let y = 0; y < tmsZoom.matrixHeight; y++) {
                 const promise = this.q(() => {
                     const tile = { x, y, z: zoom };
+                    const bounds = tms.tileToSourceBounds(tile);
+                    extent = extent == null ? bounds : extent.union(bounds);
                     const tileId = TileMatrixSet.tileToName(tile);
                     return this.renderTile(tile, logger.child({ index: tileId }));
                 });
                 promises.push(promise);
             }
         }
-        await Promise.all(promises).catch((err) => logger.fatal(err, 'FailedToRun'));
+        try {
+            this.createMetadata(extent!, await Promise.all(promises));
```

Comment:
best not to do a not null assertion here, as this may actually be null?

Since the extent is only done in side the promise function itself?
something like
```
await Promise.all(promises);
if (extent == null) throw ..
this.createMetadata(...)
```

Reply:
There is no code path where it could be null (unless tmsZoom.maxtrixHeight or matrixWidth) are null. The `!` is asserting that